src/foam/apploader/ModelFileDAO.js

```diff
@@ -56,21 +56,26 @@ foam.CLASS({
           };
 
           context.foam.RELATIONSHIP = function(r) {
-            var references = foam.json.references(x, r);
-
-            Promise.all(references.concat([
-              foam.package.waitForClass(r.sourceModel),
-              foam.package.waitForClass(r.targetModel)
-            ])).then(function() {
-              var obj = foam.dao.Relationship.create(r, x);
-
+            relationship = Promise.all(foam.json.references(x, r)).then(function() {
+               var obj = foam.dao.Relationship.create(r, x);
               obj.validate && obj.validate();
-              foam.package.registerClass(obj);
```

Comment:
You appear to have removed all of the foam.package.registerClass calls.  Do relationships get registered at all now in the global context?  There may be code that still references them that way, so you may want to still call registerClass in initRelationship()